atlasdb-dbkvs/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTableTest.java

```diff
@@ -225,25 +218,158 @@ public void dropTablesDoesNotThrowIfMultipleMatchingTableReferencesExistAndCaseS
+        // Not all tests will call OracleDdlTable#createTable, which makes sense as this test "creates" it for them!
+        lenient()
+                .when(tableNameGetter.generateShortTableName(connectionSupplier, TEST_TABLE))
+                .thenReturn(INTERNAL_TABLE_NAME);
         when(tableNameGetter.getPrefixedTableName(TEST_TABLE)).thenReturn(PREFIXED_TABLE_NAME);
         when(tableNameGetter.getInternalShortTableName(connectionSupplier, TEST_TABLE))
                 .thenReturn(INTERNAL_TABLE_NAME);
+        when(sqlConnection.selectExistsUnregisteredQuery(
+                        eq("SELECT 1 FROM "
+                                + TABLE_MAPPING_DEFAULT_CONFIG.metadataTable().getQualifiedName()
+                                + " WHERE table_name = ?"),
+                        eq(TEST_TABLE.getQualifiedName())))
+                .thenReturn(true);
     }
 
     private void createOverflowTable() throws TableMappingNotFoundException {
         when(tableNameGetter.getPrefixedOverflowTableName(TEST_TABLE)).thenReturn(PREFIXED_OVERFLOW_TABLE_NAME);
         when(tableNameGetter.getInternalShortOverflowTableName(connectionSupplier, TEST_TABLE))
                 .thenReturn(INTERNAL_OVERFLOW_TABLE_NAME);
+        when(sqlConnection.selectExistsUnregisteredQuery(
+                        eq("SELECT 1 FROM user_tables WHERE TABLE_NAME = ?"),
+                        eq(INTERNAL_OVERFLOW_TABLE_NAME.toUpperCase())))
+                .thenReturn(true);
+    }
+
+    private void createTableAndOverflow() throws TableMappingNotFoundException {
+        createTable();
+        createOverflowTable();
     }
 
     private void verifyTableDeleted(String tableName) {
         verify(sqlConnection).executeUnregisteredQuery("DROP TABLE " + tableName + " PURGE");
     }
+
+    private byte[] createMetadata(boolean hasOverflow) {
+        return TableMetadata.builder()
+                .singleNamedColumn("foo", "foobar", hasOverflow ? ValueType.STRING : ValueType.FIXED_LONG)
+                .build()
+                .persistToBytes();
+    }
+
+    private void setTableToHaveOverflowColumn(boolean hasColumn) {
+        when(sqlConnection.selectExistsUnregisteredQuery(
+                        eq("SELECT 1 FROM user_tab_cols WHERE TABLE_NAME = ? AND COLUMN_NAME = 'OVERFLOW'"),
+                        eq(INTERNAL_TABLE_NAME.toUpperCase())))
+                .thenReturn(hasColumn);
+    }
+
+    private void verifyNumberOfTimesTableAltered(int numberOfTimes) {
+        verify(sqlConnection, times(numberOfTimes))
+                .executeUnregisteredQuery("ALTER TABLE " + INTERNAL_TABLE_NAME + " ADD (overflow NUMBER(38))");
+    }
+
+    private OracleDdlTable createOracleDdlTable(OracleDdlConfig config) {
+        return OracleDdlTable.create(
+                TEST_TABLE, connectionSupplier, config, tableNameGetter, tableValueStyleCache, executorService);
+    }
+
+    private void setMetadataToHaveOverflow(boolean hasOverflow) {
```

Comment:
nit: This name admittedly confused me - we're not setting the metadata in the metadata table, but we're modifying the table value style cache, which doesn't use the metadata table. Consider renaming the method to something that reflects that (say `setTableValueStyleCacheOverflowConfigForTable`)